CopilotKit/packages/react-core/src/hooks/use-copilot-action.ts

```diff
@@ -158,7 +158,8 @@ export function useCopilotAction<const T extends Parameter[] | [] = []>(
 ): void {
   const { setAction, removeAction, actions, chatComponentsCache } = useCopilotContext();
   const idRef = useRef<string>(randomId());
-  const renderAndWaitRef = useRef<RenderAndWaitForResponse | null>(null);
+  // Replace single ref with Map to track multiple instances
+  const renderAndWaitMapRef = useRef<Map<string, RenderAndWaitForResponse>>(new Map());
```

Comment:
"// Replace single ref with Map to track multiple instances" suggest a diff from previous implementation. Instead, let's just explain the situation without previous implementation context